packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -364,12 +321,23 @@ function parseChildren(
             } else if (parse5Utils.isTextNode(child)) {
                 const textNodes = parseText(ctx, child);
                 parent.children.push(...textNodes);
+                // Non whitespace text nodes interrupt any context we may be carrying
+                if (textNodes.length > 0) {
+                    ctx.clearSiblingScope();
+                }
             } else if (parse5Utils.isCommentNode(child)) {
                 const commentNode = parseComment(child);
                 parent.children.push(commentNode);
+                // Comment behavior depends on whether preserveComments is enabled.
+                // If it is enabled, comments become syntactically meaningful and
+                // interrupt any context we may be carrying
+                if (ctx.preserveComments) {
+                    ctx.clearSiblingScope();
+                }
             }
         });
     }
+    ctx.beginSiblingScope();
```

Comment:
Should this be ctx.clearSiblingScope()?

Reply:
Yes, that's a memory leak waiting to happen.